base/compiler/abstractinterpretation.jl

```diff
@@ -2222,22 +2225,22 @@ function abstract_eval_value_expr(interp::AbstractInterpreter, e::Expr, vtypes::
                 f = abstract_eval_value(interp, stmt.args[1], vtypes, sv)
                 if f isa Const && f.val === getfield
                     # boundscheck of `getfield` call is analyzed by tfunc potentially without
-                    # tainting :inbounds or :consistent when it's known to be nothrow
+                    # tainting :noinbounds or :noub when it's known to be nothrow
                     @goto delay_effects_analysis
                 end
             end
             # If there is no particular `@inbounds` for this function, then we only taint `:noinbounds`,
-            # which will subsequently taint `:consistent`-cy if this function is called from another
+            # which will subsequently taint `:noub` if this function is called from another
             # function that uses `@inbounds`. However, if this `:boundscheck` is itself within an
             # `@inbounds` region, its value depends on `--check-bounds`, so we need to taint
-            # `:consistent`-cy here also.
+            # `:noub`-cy here also.
```

Comment:
Just took another look at this comment, and it got me wondering. Why are we checking if the current statement is inside the `@inbounds` region and then tainting it with ``:noub`? Isn't necessary now as we separate caches based on `--check-bounds`?

Also, if we do need to taint it, probably we should taint `:consistent` instead?